private/buf/bufmigrate/migrator.go

```diff
@@ -486,6 +491,10 @@ func (m *migrator) getCommitIDToCommit(
 	ctx context.Context,
 	moduleKeys []bufmodule.ModuleKey,
 ) (map[uuid.UUID]bufmodule.Commit, error) {
+	// We need to dedupe all the module keys of the configured dependencies, since in `v1`
+	// workspaces, each module will have their own configured dependencies, which may be
+	// duplicated across modules.
+	moduleKeys = dedupeStringerSlice(moduleKeys)
```

Comment:
We're deduping based on previously-seen string value for `ModuleKeys`? We have extensive logic in the `ModuleSetBuilder` relating to what `ModuleKeys` take priority based on timestamp, why is this sufficient here?

Reply:
See comment below - but basically to build the configured deps list in our `buf.yaml` configuration, we then have additional logic to handle which to choose (re: the `label1` vs `label2` example). This is only to ensure that we do not call the exact same configured reference multiple times using our providers.